src/promptflow/promptflow/_sdk/_service/run.py

```diff
@@ -0,0 +1,72 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from dataclasses import asdict
+
+from flask import Blueprint, jsonify, request
+
+from promptflow._sdk._constants import get_list_view_type
+from promptflow._sdk._service.utils import api_wrapper
+from promptflow._sdk.operations._local_storage_operations import LocalStorageOperations
+from promptflow._sdk.operations._run_operations import RunOperations
+from promptflow.contracts._run_management import RunMetadata
+
+run_bp = Blueprint("run", __name__, url_prefix="/run/v1.0")
+
+
+@run_bp.route("/list", methods=["GET"])
+@api_wrapper
+def list():
+    # parse query parameters
+    max_results = request.args.get("max_results", default=50, type=int)
+    all_results = request.args.get("all_results", default=False, type=bool)
```

Comment:
generalize max_result and all_results dealing?

Reply:
by the way, I don't see --stream support in your patch?

Reply:
offline synced:

1. connection and run have a little bit different behavior on list operation, and for now there is no third list operation, so not generalize in this PR.
2. stream is not in this PR's scope (also, there is no ask for this)